osmosis-labs/osmosis-frontend
# 🤗 Embrace the 1-Click: Add 1-Click enable and modify to swap review

~~⚠️ Note: An infinite rerender issue exists when an amount is entered into the swap input. This was introduced in https://github.com/osmosis-labs/osmosis-frontend/pull/3949 and will affect this PR~~ Solved by #3956 

## What is the purpose of the change:

Adding the 1 click trading enable and parameter settings to the review order modal to promote the feature.

<details>
<summary>Design from figma</summary>

![CleanShot 2024-11-14 at 17 34 58](https://github.com/user-attachments/assets/cd87fff2-48d4-4eac-9c51-667eeb08a1b6)

</details>

## Notes/TBD
- ~~After enabling 1CT through the `ReviewOrder` modal the actual swap needs to be signed as well~~
  - ~~this is a bit quirky, but it seemed better than abusing our newfound power and executing a more expensive swap that was shown to the user (even though the usd amount of the change is negligible)~~
- The profile settings could be reworked so that part also uses `useOneClickTradingSessionManager` but wanted to keep the existing code paths for now
  - this feature also could be a good fit for using Zustand but didn't want to do that without buy-in

## Brief Changelog

- reworked the parameter setting modal to support "in flow" usage 
  - this need wasn't explicitly stated in the requirements, but keeping the existing behavior would have contradicted some user stories
- reworked `useOneClickTradingSession` and `useOneClickTradingParams` to properly propagate changes to the 1CT session coming from the account store
- extend `ReviewOrder` modal to implement the design and features
- create `useOneClickTradingSessionManager` for managing states 
- made the trpc logs configurable
- added short time units 

the commit messages are fairly well structured and descriptive

## Testing and Verifying

This change has been tested locally by rebuilding the website and verified content and links are expected

- enable 1ct through the reivew order modal
  - play with changing + saving/discarding the config
- when 1ct check is enabled, modify the params and approve the changes
- check the profile state shows the proper settings
- change 1ct in through the profile menu and check review order modal
- set the spending limit to 0usd and check the `Exceeded` state 

More nuanced things can be tested, but these cover the main paths. 

## Demo
![CleanShot 2024-11-14 at 17 10 34](https://github.com/user-attachments/assets/8fc3b2c4-d34f-451e-af65-9fad049b80e3)
